spicy/toolchain/src/compiler/parser/scanner.ll

```diff
@@ -231,7 +232,7 @@ Null                  return token::CNULL;
 <NEW_KEYWORD>skip     return token::SKIP;
 
 {attribute}           yylval->build(std::string(yytext)); return token::ATTRIBUTE;
-<INITIAL>{property}   yylval->build(std::string(yytext)); return token::PROPERTY;
+{property}            yylval->build(std::string(yytext)); return token::PROPERTY;
```

Comment:
This change could break existing code, e.g.,

```ruby
module foo;

const debug = 2;

public type X = unit {
    var did_run: bool = False;

    xs: uint8[] &size=3 %debug {
        self.did_run = True;
        assert |$$| == 1: "expected 1, got %d" % |$$| ;
    }

    on %done {
        assert self.did_run;
    }
};
```

Before this would behave identical for debug and non-debug builds since the constant was used, now we interpret this as a hook which should only run in debug mode.

I do not have a good idea whether it is possible to fix the parsing with error hooks starting with `%error`. Another option might be to at least switch the inline form to `error` (which would be consistent with `foreach` 😬).

Reply:
Yes, but this is a more general problem, our (existing) syntax is simply ambiguous, and I don't see how we can fix it 100%. We had the same issue with attributes and have fixed it the same way as I do here. I can live with `&size=3 %debug` being interpreted the wrong way if somebody actually wants that as an expression there (i.e., (1) have an ID called `debug` that supports modulo and (2) skip the space when typing out that modulo expression).

Reply:
Sounds good, and should be okay if we call that out as breaking changes in the release notes.